template/lambda-sqs-worker-cdk/src/framework/handler.ts

```diff
@@ -23,18 +29,46 @@ const withDatadog = <Event, Output = unknown>(
 export const createHandler = <Event, Output = unknown>(
   fn: (event: Event, ctx: LambdaContext) => Promise<Output>,
 ) =>
-  withDatadog<Event>((event, ctx) =>
-    loggerContext.run({ awsRequestId: ctx.awsRequestId }, async () => {
-      try {
-        const output = await fn(event, ctx);
+  withDatadog<Event>(async (event, ctx) => {
+    try {
+      const output = await fn(event, ctx);
 
-        logger.debug('Function succeeded');
+      logger.debug({ awsRequestId: ctx.awsRequestId }, 'Function completed');
 
```

Comment:
Rephrased to `completed` as this could still complete while returning batch failures.